components/style/values/computed/mod.rs

```diff
@@ -169,3 +170,73 @@ impl ToCss for BorderRadiusSize {
 
 pub type Number = CSSFloat;
 pub type Opacity = CSSFloat;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum AngleOrCorner {
+    Angle(Angle),
+    Corner(HorizontalDirection, VerticalDirection),
+}
+
+impl ToComputedValue for specified::AngleOrCorner {
+    type ComputedValue = AngleOrCorner;
+
+    #[inline]
+    fn to_computed_value(&self, _: &Context) -> AngleOrCorner {
+        match *self {
+            specified::AngleOrCorner::Angle(angle) => {
+                AngleOrCorner::Angle(angle)
+            },
+            specified::AngleOrCorner::Corner(horizontal, vertical) => {
+                match (horizontal, vertical) {
+                    (None, Some(VerticalDirection::Top)) =>
+                        AngleOrCorner::Angle(Angle(0.0)),
+                    (Some(HorizontalDirection::Right), Some(VerticalDirection::Top)) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.25)),
+                    (Some(HorizontalDirection::Right), None) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.5)),
+                    (Some(HorizontalDirection::Right), Some(VerticalDirection::Bottom)) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.75)),
+                    (None, Some(VerticalDirection::Bottom)) =>
+                        AngleOrCorner::Angle(Angle(PI)),
+                    (Some(HorizontalDirection::Left), Some(VerticalDirection::Bottom)) =>
```

Comment:
If both vertical and horizontal position is specified, we need to store them as keywords instead of converting them to angle(like deleted specified parsing part)